qutebrowser/browser/urlmarks.py

```diff
@@ -218,6 +215,24 @@ def quickmark_del(self, name):
         except KeyError:
             raise cmdexc.CommandError("Quickmark '{}' not found!".format(name))
 
+    def get_by_qurl(self, url):
+        """Look up a quickmark by QUrl, returning its name.
+
+        Takes O(n) time, where n is the number of quickmarks.
+        Use a name instead where possible.
+        """
+        if not url.isValid():
+            raise ValueError("Invalid URL: {}".format(url.errorString()))
+        urlstr = url.toString(QUrl.RemovePassword | QUrl.FullyEncoded)
+
+        try:
+            index = list(self.marks.values()).index(urlstr)
+            key = list(self.marks.keys())[index]
```

Comment:
Doing anything serious for performance is probably overkill since people generally shouldn't have thousands of quickmarks (and if you do, just delete by name or edit the file).

As I said in the PR description, this just deletes one quickmark, chosen in Python order. That should be alphabetical by name if the quickmarks haven't changed since loading them, but otherwise could be anything.